### PR TITLE
adding dark theme to chart

### PR DESCRIPTION
**Pull request by shayanbinary**
@brucebinary  Can you please check this PR,
I want to be sure I am in a right way

### CODE REVIEW

**Comment by brucebinary** [comment]
- I think you need to have a `clearStyles` when you set a new theme for the chart to clear the old CSS for canvas. Refer http://documentation.chartiq.com/tutorial-CSS%20Overview.html#main
 - binary-static will be passing all chart settings in `ChartSetting.jsx` as prop. Implement those as well, and have the prop values have precedence over those in local storage (`lang` is already doing this). Assume that if prop setting is passed, user will not use `ChartSetting.jsx`.

**Comment by brucebinary** [comment]
@shayanbinary Also once you added those props, add the documentation in README under #API. 

Keep up the good work! (:

File: sass/_variables.scss [hunk 1]
```diff
@@ -45,6 +46,60 @@ $DEFAULT-border:          #e9e9ed !default;
...
+$control-active:          #4b5794 !default;
+$control-bg-active:       rgba(76, 87, 148, 0.2) !default;
+$control-grid:            #fafafb !default;
+$control-dialog-head:     #FFFFFF !default;
+$control-dialog-body:     #FAFAFA !default;
+$control-title-bg:        #FAFAFA !default;
+
+
+
+/******************************\
```
Comment (brucebinary) [inline sass/_variables.scss]: there's no light/dark theme in variables anymore right?

File: sass/styles/_menu.scss [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+				@include svg-icon-color('activeColor');
+				.ic-subtitle{
+					font-weight: 500;
+					@include themify($themes) {
+						color: themed('controlTextActive');
+					}
+				}
+			}
+		}
+		// > .ic-icon,
```
Comment (brucebinary) [inline sass/styles/_menu.scss]: if it not used just remove it, otherwise document what uncommenting will do

File: src/components/Chart.jsx [hunk 3]
```diff
@@ -63,9 +63,18 @@ class Chart extends Component {
         const insideHolder = array.filter(c => !/(TradeStart)|(TradeEnd)/.test(c.type.displayName));
         const insideSubHolder = array.filter(c => /(TradeStart)|(TradeEnd)/.test(c.type.displayName));
         const renderTopWidgets = topWidgets || defaultTopWidgets;
+        const contextClassName = () =>{
```
Comment (brucebinary) [inline src/components/Chart.jsx]: this function can be simplified. In fact you may not even need it with template literals

File: src/store/ChartSettingStore.js [hunk 4]
```diff
@@ -116,10 +120,14 @@ export default class ChartSettingStore {
...
+        this.theme = value ? 'dark' : 'light';
+        this.mainStore.chart.stxx.clearStyles();
+        this.saveSetting();
+    }
 
-
-    @action.bound setView(view) {
-        this.view = view ? view : '';
+    @computed get isDarkTheme() {
+        return this.theme === 'dark';
```
Comment (brucebinary) [inline src/store/ChartSettingStore.js]: this is just nitpicking but... it's a one line function that's only used in one place. You could just pass this condition to the `<Switch />`

Comment (brucebinary) [reply src/store/ChartSettingStore.js]: @shayanbinary actually never mind this.

File: sass/components/_categorical-display.scss [hunk 5]
```diff
@@ -1,310 +1,359 @@
...
+        line-height: 3.4em;
+        text-align: left;
+        cursor: pointer;
+        @include transition(background-color 0.3s)
+        @include box-sizing(border-box);
+        &.cq-active-filter {
+            font-weight: 800;
+            @include themify($themes) {
+                color: themed('defaultTextActive');
+                border-right: 2px solid themed('activeColor');
```
Comment (aminroosta) [inline sass/components/_categorical-display.scss]: @shayanbinary Could you please check if `themed(...)` can be used in this way.
I may be wrong but i think `property: value;` syntax dictates `themed(...)` to be the only value.

If that is the case, the fix is obvious
```
border-right-width: 2px;
border-right-style: solid;
@include themify($themes) {
     border-right-color: themed('activeColor');
}
```

Again, it's just an skepticism, let me know if the above works the way it is!

File: sass/components/_categorical-display.scss [hunk 6]
```diff
@@ -1,310 +1,359 @@
...
+    .cq-item, .cq-active-item {
+        padding-left: 20px;
+        padding-right: 20px;
+        display: flex;
+        height: 40px;
+        align-items: center;
+        justify-content: space-between;
+        line-height: initial;
+        @include themify($themes) {
+            border-bottom: 1px solid themed('defaultBorder');
```
Comment (aminroosta) [inline sass/components/_categorical-display.scss]: same here.

File: sass/components/_categorical-display.scss [hunk 7]
```diff
@@ -1,310 +1,359 @@
...
-                color: $active-color;
-                font-weight: bold;
-                padding: 0.78em 1em;
-                width: 100%;
-                cursor: pointer;
-                @include box-shadow(0,0,0,#fff);
+            cursor: pointer;
+            @include themify($themes) {
+                color: themed('activeColor');
+                @include box-shadow(0,0,0,themed('defaultMain'));
```
Comment (aminroosta) [inline sass/components/_categorical-display.scss]: same.

File: sass/components/_categorical-display.scss (hunk 5 above)
Comment (shayanbinary) [reply sass/components/_categorical-display.scss]: @aminroosta  this piece of code fell me in love with `sass-loader` plugin. to be sure I make  sample on `http://shayanbinary.binary.sx/theme-test/` please click on `chart setting` and change theme. I use bellow code
` @include themify($themes) {
     border: 2px solid themed('activeColor');
}`

**Pull request merged**